runtime-testsuite/test/org/antlr/v4/test/runtime/BaseRuntimeTest.java

```diff
@@ -426,35 +420,39 @@ public static ErrorQueue antlrOnString(String workdir,
 	public static UniversalRuntimeTestDescriptor readDescriptor(String dtext)
 			throws RuntimeException
 	{
-		String[] fileSections = {
-				"notes","type","grammar","slaveGrammar","start","input","output","errors",
-				"flags","skip"};
-		Set<String> sections = new HashSet<>(Arrays.asList(fileSections));
 		String currentField = null;
-		String currentValue = null;
+		StringBuilder currentValue = new StringBuilder();
 
 		List<Pair<String, String>> pairs = new ArrayList<>();
-		String[] lines = dtext.split("\r?\n");
-		for (String line : lines) {
-			if ( line.startsWith("[") && line.length()>2 &&
-				 sections.contains(line.substring(1, line.length() - 1)) ) {
-				if ( currentField!=null ) {
-					pairs.add(new Pair<>(currentField,currentValue));
+
+		Matcher matcher = linePattern.matcher(dtext);
+		boolean preserveSeparator = false;
+		int lastEnd = 0;
+		while (matcher.find()) {
+			String line = matcher.group(1);
+			lastEnd = matcher.end();
+			boolean newSection = false;
+			String sectionName = null;
+			if (line.startsWith("[") && line.length() > 2) {
+				sectionName = line.substring(1, line.length() - 1);
+				newSection = sections.contains(sectionName);
+			}
+
+			if (newSection) {
+				if (currentField!=null) {
+					pairs.add(new Pair<>(currentField, currentValue.toString()));
 				}
-				currentField = line.substring(1, line.length() - 1);
-				currentValue = null;
+				preserveSeparator = sectionName.equals("input");
```

Comment:
i see; you needed to upgrade code to handle test files with various line terminators.